### ISSUE TITLE
b-card title is always blank

### ISSUE THREAD

**Issue by WayneHiller** [bootstrap-vue/bootstrap-vue#4092]
Since updating to bootstrap-vue 2.0 my b-card titles are all blank. 

![image](https://user-images.githubusercontent.com/1186731/64895364-33a98500-d631-11e9-821e-043c82327a0a.png)

**Comment by tmorehouse** [comment]
Could you provide a sample of your pages markup?

**Comment by tmorehouse** [comment]
When I use the following markup:

```html
  <b-card title="Some Title">
    <b-form-row>
      <b-form-input></b-form-input>
    </b-form-row>
  </b-card>
```

I see:

![image](https://user-images.githubusercontent.com/2781561/64897710-beeb4080-d65a-11e9-8c85-937906b34dd1.png)

**Comment by WayneHiller** [comment]
That is what I would expect. I am working on creating a little demo app to see if I can reproduce it. Note that I also changed my entire project to use typescript with component classes as well.

**Comment by WayneHiller** [comment]
bootstrap-vue 2.0.1
vue 2.6.10

This is so crazy. Looking at the code the only way I can see that the h2 would be empty is if children was assigned but that should not be possible.

I built a little test app and it works fine.
```
// @vue/component
export const BCardTitle = /*#__PURE__*/ Vue.extend({
  name: 'BCardTitle',
  functional: true,
  props,
  render(h, { props, data, children }) {
    return h(
      props.titleTag,
      mergeData(data, {
        staticClass: 'card-title'
      }),
      children || props.title
    )
  }
```
The same thing is happening with the sub-title
```
<b-card title="Title" 
         sub-title="SubTitle"
          header="Header"
          footer="Footer">
      content
</b-card>
```
![image](https://user-images.githubusercontent.com/1186731/64899755-ceaa5b00-d641-11e9-9b58-c0e0773c252f.png)

**Comment by WayneHiller** [comment]
Could something be going wrong in the call to plunkProps maybe? I am grasping at straws here lol.

```
// @vue/component
export const BCardBody = /*#__PURE__*/ Vue.extend({
  name: 'BCardBody',
  functional: true,
  props,
  render(h, { props, data, children }) {
    let cardTitle = h()
    let cardSubTitle = h()
    const cardContent = children || [h()]

    if (props.title) {
      cardTitle = h(BCardTitle, { props: pluckProps(titleProps, props) })
    }
```

**Comment by tmorehouse** [comment]
Unless vue-component-class is having issues with child functional components

**Comment by WayneHiller** [comment]
@tmorehouse I was thinking the same thing so I added a test page like this, same issue.

```
<template>
    <div>
        <h1 class="text-center">AIS Manager</h1>
        <b-card title="Title" 
                sub-title="SubTitle"
                header="Header"
                footer="Footer">
            content
        </b-card>
    </div>
</template>

<script>
    export default {
        name: "index"
    }
</script>

```

**Comment by WayneHiller** [comment]
WOW I found it.  In one component I forgot the Component decorator.

```
<script lang="ts">
    import { Component, Vue, Prop, Ref, Watch } from "vue-property-decorator";

   --> Missing Component here
    export default class CoreDealerSentMsgsChart extends Vue {
```
I found it because some how that component replaced the Root in the Vue component list.
![image](https://user-images.githubusercontent.com/1186731/64911915-b1769a80-d6dc-11e9-93a0-585cf7b5c419.png)

Going to need to work with the vue-component-class project to find a way to detect this issue. Amazing that the app even ran really.  YAY puzzle solved :)
